### ISSUE TITLE
Merging CARMA aerosol model for troposphere and stratosphere (trop_strat) into the CESM development version

### ISSUE THREAD

**Issue by tilmes** [ESCOMP/CAM#495]
CARMA is the Community Aerosol and Radiation Model for the Atmosphere, which is a framework for modeling both aerosols and clouds using a size resolving description. While there are already some CARMA models linked to CESM2, there is currently no version that completely replaces the modal aerosols in CAM with the sectional aerosol model in both the troposphere and the stratosphere. Currently, existing aerosol models in CESM2 run in parallel with the MAM scheme and do not interact fully with clouds, radiation, and chemistry.

The newly developed sectional aerosol model has the advantage to describe aerosol microphysics in much more detail than a modal aerosol model and is more physical because it fully integrates the sectional aerosols in physics, radiation, and chemistry. The implementation of a model is in particular important for exploring the effects of stratospheric aerosols injection (geoengineering), but also to understand the effects of volcanoes. The coupling to the entire climate system is therefore needed. In addition, for the troposphere, this model has been shown to be able to resolve tropospheric aerosols well. A presentation to the CGD process group on the topic was given in May 2021. 
[CGD_process_meeting_May2021.pptx](https://github.com/ESCOMP/CAM/files/7831331/CGD_process_meeting_May2021.pptx)

The development has been performed in an earlier model version (close to CESM2.2.0) and needs to be merged into the development code.  The implementation includes changes and additions of several routines that are listed in this google doc: https://docs.google.com/document/d/15if8UXjDcyhduCjOCVOgU5mHsoXIIepHI0CPzuj5k2g/edit
Details of the different changes in the different routines can be provided.

**Comment by fvitt** [comment]
See branch:
https://github.com/fvitt/CAM/tree/carma_model_dev
